#####################################################
#                                                   #
#               ObeseMiceDeSEQ2.R                   #
#               James Wengler - jwengler@gmail.com  #
#               Last Modified: June 21, 2022        #
#                                                   #
#####################################################
# This file runs a DESeq2 analysis (differential gene expression) on the mice obesity RNASeq Data
# REQUIRED: count table, meta table generated by cleanCount.py
# OUTPUT: whatever DESeq2 analysis you desire



# Package that runs the actual differentl
library(DESeq2)
library(tidyverse)

# This path should point at the count table
countData <- read.csv("/Users/jameswengler/final_count_table.csv", row.names = 1)
# This path should point to the metadata table
metaData <- read.csv("/Users/jameswengler/meta_count_table.csv")

# This command can be broken done into 3 parts
# 1. pass the count table to the varible countData
# 2. pass the meta table to the variable colData
# 3. The deign variable will be HIGHLY dependent on what you want to do. A quick google search
# will help you decide what to set this variable as
dds <- DESeqDataSetFromMatrix(countData = countData, colData = metaData, design = ~treatment)

# Analysis below! This can be a traditional DESeq analysis, a PCA graph, whatever you decide
# This is a great resource for looking at the various things DESeq2 can do: https://lashlock.github.io/compbio/R_presentation.html
dds <- DESeq(dds)
#res <- results(dds)
vsdata <- vst(dds, blind=FALSE)
plotPCA(vsdata, intgroup = "treatment")